src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/MSBuildXmlAccessor.cs

```diff
@@ -31,10 +31,8 @@ internal class MSBuildXmlAccessor : IProjectXmlAccessor
         {
             using (var access = await _projectLockService.ReadLockAsync())
             {
-                var stringWriter = new EncodingStringWriter(await _unconfiguredProject.GetFileEncodingAsync().ConfigureAwait(true));
                 var projectXml = await access.GetProjectXmlAsync(_unconfiguredProject.FullPath).ConfigureAwait(true);
-                projectXml.Save(stringWriter);
-                return stringWriter.ToString();
+                return projectXml.RawXml;
```

Comment:
`ProjectRootElement.Save` has the side-effect of clearing the dirty flag, which was causing the bug. RawXml does not have this side-effect.